web_group_expand/static/src/js/web_group_expand.js

```diff
@@ -0,0 +1,74 @@
+odoo.define('web_groupby_expand.web_groupby_expand', function (require) {
+"use strict";
+
+var ViewManager = require('web.ViewManager');
+
+ViewManager.include({
+    render_view_control_elements: function (){
+        var res = this._super.apply(this, arguments);
+        if (this.searchview_elements) {
+            var searchview = this.searchview_elements.$searchview_buttons
+            var expand_button = searchview.find('#oe_group_by_expand');
+            var reset_button = searchview.find('#oe_group_by_reset');
+            expand_button.on('click', this.proxy('expand_records'));
+            reset_button.on('click', this.proxy('reset_records'));
+            reset_button.hide();
+            expand_button.hide();
+        }
+        return res;
+    },
+
+    _process_search_data: function () {
+        var res = this._super.apply(this, arguments);
+        if (this.active_view && this.active_view.type == 'list' && this.searchview_elements) {
+            var searchview = this.searchview_elements.$searchview_buttons
+            var expand_button = searchview.find('#oe_group_by_expand');
+            var reset_button = searchview.find('#oe_group_by_reset');
+
+            if (res.groupBy.length > 0) {
+                reset_button.show()
+                expand_button.show()
+            }
+            else {
+                reset_button.hide()
+                expand_button.hide()
+            }
```

Comment:
You can create a new method for ex. do_toggle_visibility instead of calling  hide/show 3 times.

Reply:
Previously i made some changes in button template, so, could i assume that you propose something like this ?:
```
render_view_control_elements: function (){
// ...
var has_groups = res.groupBy.length > 0
this.do_toggle_visibility(has_groups)
// ...
}

function do_toggle_visibility(show) {
    var searchview.find('.toggle_buttons');
    if (show) {
        toggle_buttons.show()
    }
    else {
        toggle_buttons.hide()
    }
}
```

Reply:
I liked, changes made